world/physics/RigidBody.h

```diff
@@ -0,0 +1,127 @@
+#ifndef C___RIGIDBODY_H
+#define C___RIGIDBODY_H
+
+#include "math/EigenWrapper.h"
+
+StateVector f(const StateVector& x, const Vector6& u, const Matrix_3x3& InertiaTensor,
+              const Matrix_3x3& InertiaTensorInverse, double mass);
+
+StateVector rk4(const StateVector& x, const Vector6& u, const Matrix_3x3& InertiaTensor,
+                const Matrix_3x3& InertiaTensorInverse, double mass, double dt);
+
+/**
+ * @brief Assumes body-frame origin coincides with the center-of-mass
+ *
+ */
+class RigidBody {
+   public:
+    /**
+     * @brief Construct a new Rigid Body object
+     *
+     * @param mass // [kg]
+     * @param InertiaTensor in body frame // [kg * m^2]
+     * @param init_pos_b_wrt_g_in_g // [m]
+     * @param init_g_q_b initial rotation
+     * @param init_vel_b_wrt_g_in_b // [m/s]
+     * @param init_omega_b_wrt_g_in_b // [rad/s]
+     */
+    RigidBody(double mass, const Matrix_3x3& InertiaTensor, const Vector3& init_pos_b_wrt_g_in_g,
+              const Quaternion& init_g_q_b, const Vector3& init_vel_b_wrt_g_in_b,
+              const Vector3& init_omega_b_wrt_g_in_b);
+
+    /**
+     * @brief Compute force due to gravity, expressed in body frame
+     *
+     * @return Vector3 [N]
+     */
+    Vector3 get_gravity_b();
+
+    Vector3 get_net_force_b();
+    Vector3 get_net_moment_b();
+    Vector3 get_pos_b_wrt_g_in_g();
+    Quaternion get_g_q_b();
+    Quaternion get_b_q_g();
+    Vector3 get_vel_b_wrt_g_in_b();
+    Vector3 get_omega_b_wrt_g_in_b();
+
+    /**
+     * @brief Applies a force
+     *
+     * @param force_b force to apply [N]
+     * @param pointOfApplication_b body position at which to apply the force // [m]
+     */
+    void applyForce(const Vector3& force_b, const Vector3& pointOfApplication_b);
+
+    /**
+     * @brief Applies a moment
+     *
+     * @param moment_b moment to apply [N*m]
+     */
+    void applyMoment(const Vector3& moment_b);
+
+    void clearAppliedForcesAndMoments();
+
+    /**
+     * @brief Fourth order Runge-Kutta integration
+     *
+     * @param dt integration time [s]
+     */
+    void rk4(double dt);
```

Comment:
Is this method being used? There's no definition in the corresponding cpp file

Reply:
Deleted class definition and this method along with it